packages/@vue/cli-service/lib/config/app.js

```diff
@@ -214,15 +214,20 @@ module.exports = (api, options) => {
         const entries = Array.isArray(entry) ? entry : [entry]
         webpackConfig.entry(name).merge(entries.map(e => api.resolve(e)))
 
+        // test whether use inline loader
+        // * See https://github.com/jantimon/html-webpack-plugin/blob/master/docs/template-option.md#2-setting-a-loader-directly-for-the-template
+        const inlineLoaderReg = /[-!]?!.+!/
+        const isWithInlineLoader = /^[-!]?!/.test(template)
+
         // resolve page index template
-        const hasDedicatedTemplate = fs.existsSync(api.resolve(template))
+        const hasDedicatedTemplate = fs.existsSync(api.resolve(isWithInlineLoader ? template.replace(inlineLoaderReg, '') : template))
```

Comment:
I think the `isWithInlineLoader` and `inlineLoaderReg` is not very valid. e.g. it can't match path without prefix `!!`  `handlebars-loader!src/index.hbs`

`const templateWithoutLoaders = templateFilename.replace(/^.+!/, '').replace(/\?.+$/, '');` 

> https://github.com/jantimon/html-webpack-plugin/blob/0a6568d587a82d88fd3a0617234ca98d26a1e0a6/index.js#L129

Reply:
done